### PR TITLE
added: IBatch Interface

### LINKED ISSUE #325: Implement `IBatch` interface
## To do

- [x] Implement `IBatch` interface
- [x] Inherit `IBatch` into `ISablierFlowBase`

### PR DESCRIPTION
**Pull request by TheMarvelFan**
Closes #325 

Added IBatch interface and inherited it to ISablierFlowBase inerface.
Also inherited IBatch into Batch to fix inheritance issues during compile-time.

### CODE REVIEW

File: src/abstracts/Batch.sol [hunk 1]
```diff
@@ -2,18 +2,20 @@
 pragma solidity >=0.8.22;
 
 import { Errors } from "../libraries/Errors.sol";
+import { IBatch } from "../interfaces/IBatch.sol";
 
 /// @title Batch
 /// @notice This contract implements logic to batch call any function.
```
Comment (andreivladbrg) [inline src/abstracts/Batch.sol:8]: we should say for the `notice` natpsec:

```solidity
/// @notice See the documentation in {IBatch}.
```

File: src/abstracts/Batch.sol [hunk 2]
```diff
@@ -2,18 +2,20 @@
...
+abstract contract Batch is IBatch {
     /*//////////////////////////////////////////////////////////////////////////
                          USER-FACING NON-CONSTANT FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
+    /// @inheritdoc IBatch
     /// @notice Allows batched call to self, `this` contract.
     /// @param calls An array of inputs for each call.
-    function batch(bytes[] calldata calls) external {
+    function batch(bytes[] calldata calls) external virtual override {
```
Comment (andreivladbrg) [inline src/abstracts/Batch.sol:18]: there is no need to use `virtual` here

File: src/interfaces/ISablierFlowBase.sol [hunk 3]
```diff
@@ -18,7 +18,8 @@ import { IFlowNFTDescriptor } from "./IFlowNFTDescriptor.sol";
 interface ISablierFlowBase is
     IERC4906, // 2 inherited components
     IERC721Metadata, // 2 inherited components
-    IAdminable // 0 inherited components
+    IAdminable, // 0 inherited components
+    IBatch // 0 inherited components
```
Comment (andreivladbrg) [inline src/interfaces/ISablierFlowBase.sol:22]: it should be inherited in `ISablierFlow` instead, as the `Batch` contract is not inherited in `SablierFlowBase` but in `SablierFlow`

also, the inherited components count should be updated

**Review by andreivladbrg — CHANGES_REQUESTED** [review]
Thank you for working on this, just left some comments bellow:

**Review by smol-ninja — APPROVED** [review]
LGTM. Thanks @TheMarvelFan for the PR. We use `slither` to fetch the number of inherited components.

```bash
slither src/SablierFlow.sol --print inheritance
```

**Review by andreivladbrg — APPROVED** [review]
👍

**Pull request merged**